website/1.0.1/src/site/markdown/tutorial_distributed_lock.md

```diff
@@ -0,0 +1,210 @@
+   * false if the lock is not locked by the user or the unlock operation failed
+   */
+  boolean unlock();
+
+  /**
+   * Retrieve the information of the current lock on the resource this lock object specifies, e.g. lock timeout, lock message, etc.
+   * @return lock metadata information
+   */
+  LockInfo getCurrentLockInfo();
+
+  /**
+   * If the user is current lock owner of the resource
+   * @return true if the user is the lock owner,
+   * false if the user is not the lock owner or the lock doesn't have a owner
+   */
+  boolean isCurrentOwner();
+
+  /**
+   * Call this method to close the lock's zookeeper connection
+   * The lock has to be unlocked or expired before this method can be called
+   */
+  void close();
+}
+```
+
+Currently, Helix supports two kinds of locks. One is basic lock, and the other is priority lock. We will describe their definition and usage as follows.
+
+### Helix Basic Lock
+#### Features
+* Non-blocking lock. It means every request to acquire the lock will immediately get a result (success or failure) instead of waiting until the lock is successfully acquired.
+
+* Timeout support. A client needs to input the timeout when it tries to acquire the lock. We use lazy timeout for the lock. It means even after the timeout, if no other client tries to get the lock, the client can still keep the lock, if other client tries to acquire the lock, and finds the previous client already timed out, the previous client will automatically lose the lock.
+  
+* Lock message support. A client can input the reason for the lock (as a string) when it tries to acquire the lock. This enables future lock operation, when fails, can know the reason from the lock message of the current lock. 
+  
+* Java API support for clients to use the lock.
+  
+* No notification support. When a client loses its lock after the timeout or due to some urgent maintenance work who does not check the lock, there is no notification sent out to the previous lock owner.
+
+#### How to Use Helix Basic Lock
+The implementation of Helix lock is called "ZkHelixNonblockingLock". It has two different constructors. One takes clusterName, and HelixConfigScope as inputs, while the other takes a lockPath string as input.  Note that HelixConfigScope can be "cluster", "participant", or "resource". 
+
+Based on the example code shown below, the client can use the lock in two different ways. If the client would like to use the lock for some specific Helix resource, it can use the first constructor which requires clusterName and HelixConfigScope. Or the client may choose to use the lock for generic purpose, basically on any resource they have. In this case, the client needs to provide the lock path, which is a string, and it represents where the lock should exist under zookeeper. 
+
+Note that if users have onboarded to ZooScalability that has multiple realms, they should instead use the builder instead of the above constructors.
+
+```
+/**
```

Comment:
Let's not paste the implementation here. You can attache the code github link instead.